src/interfaces/permissions/request/message.ts

```diff
@@ -0,0 +1,135 @@
+  }
+
+  get grantedTo(): string {
+    return this.message.descriptor.grantedTo;
+  }
+
+  get description(): string {
+    return this.message.descriptor.description;
+  }
+
+  get scope(): Scope {
+    return this.message.descriptor.scope;
+  }
+
+  static getType(): string {
+    return 'PermissionsRequest';
+  }
+
+  static getInterface(): string {
+    return 'Permissions';
+  }
+
+  static async create(opts: PermissionsRequestOpts): Promise<PermissionsRequest> {
+    const { conditions } = opts;
+    const providedConditions = conditions ? conditions : {};
+    const mergedConditions = { ...DEFAULT_CONDITIONS, ...providedConditions  };
+
+    const descriptor: PermissionsRequestDescriptor = {
+      conditions  : mergedConditions,
+      description : opts.description,
+      grantedTo   : opts.grantedTo,
+      grantedBy   : opts.grantedBy,
+      method      : 'PermissionsRequest',
+      objectId    : opts.objectId ? opts.objectId : uuidv4(),
+      scope       : opts.scope,
+    };
+
+    const descriptorCid = await generateCid(descriptor);
+
+    const authPayload = { descriptorCid: descriptorCid.toString() };
+    const authPayloadStr = JSON.stringify(authPayload);
+    const authPayloadBytes = new TextEncoder().encode(authPayloadStr);
+
+    const signer = await GeneralJwsSigner
+      .create(authPayloadBytes, [opts.signingMaterial]);
+
+    const message: JsonPermissionsRequest = { descriptor, authorization: signer.getJws() };
+
+    return new PermissionsRequest(message);
+  }
+
+  /**
+   * @throws {Error} if descriptorCid is missing from Auth payload
+   */
+  async verifyAuth(didResolver: DIDResolver): Promise<AuthVerificationResult> {
+    const verifier = new GeneralJwsVerifier(this.message.authorization);
+
+    // signature verification is computationally intensive, so we're going to start
+    // by validating the payload.
+
+    const payloadBytes: Uint8Array = verifier.decodePayload();
+    const payloadStr = new TextDecoder().decode(payloadBytes);
+    let payloadJson;
+
+    try {
+      payloadJson = JSON.parse(payloadStr);
+    } catch(e) {
+      throw new Error('auth payload must be a valid JSON object');
+    }
+
+    if(!isPlainObject(payloadJson)) {
+      throw new Error('auth payload must be a valid JSON object');
+    }
+
+    const { descriptorCid } = payloadJson;
+    if (!descriptorCid) {
+      throw new Error('descriptorCid must be present in authorization payload');
+    }
+
+    // parseCid throws an exception if parsing fails
+    const providedDescriptorCid = parseCid(descriptorCid);
+    const expectedDescriptorCid = await generateCid(this.message.descriptor);
+
+    if (!providedDescriptorCid.equals(expectedDescriptorCid)) {
+      throw new Error('provided descriptorCid does not match expected CID');
+    }
+
+    const { signers } = await verifier.verify(didResolver);
+
+    return { signers };
+  }
+}
+
+export const DEFAULT_CONDITIONS: Conditions = {
```

Comment:
`Conditions` feels too generic, consider perhaps:
```suggestion
export const DEFAULT_CONDITIONS: GrantConditions = {
```

Reply:
done: https://github.com/TBD54566975/dwn-sdk-js/commit/f99c98c46a5aa494a8041bea0cba32d754c3030a